CMakeLists.txt

```diff
@@ -133,4 +132,7 @@ configure_file(
 
 FILE(COPY "${CMAKE_CURRENT_BINARY_DIR}/ufs_srweather_app_meta.h" DESTINATION include)
 
-add_subdirectory(src)
+add_custom_target(mbuild DEPENDS sorc/pbuild)
+add_custom_target(minstall DEPENDS sorc/pinstall)
```

Comment:
Is this the magic that separates the build and install steps?

Are the names of the targets arbitrary? I suppose I'm curious why `mbuild` and `minstall` instead of just `build` and `install`.

Reply:
Yes, but the cause of the problem was the external projects add. You would expect a `make all` would just build the binaries and not install but default behavior of `ExternalProject_Add` is non-intiutive -- it builds and installs on a `make all`. So I added `STEP_TARGETS build install` to separate them and then connect to a custom target at the top level. I think future CMake version could potentially simplify this. I used `mbuild` and `minstall` because I couldn't use "install" as target since it is a reserved keyword. "build" was fine though but changed it to be consistent.

Reply:
I have removed the custom install target  because there was already one anyway,  so we have a `build` target instead of `mbuild` now.  Still couldn't find a solution to prevent `make all` from installing the project...